x-pack/plugins/endpoint/server/routes/resolver/utils/tree.ts

```diff
@@ -106,35 +116,37 @@ export class Tree {
       const id = entityId(event);
 
       this.ensureCache(id);
-      this.cache[id].events.push(event);
+      const currentNode = this.cache.get(id);
+      if (currentNode !== undefined) {
+        currentNode.events.push(event);
+      }
     });
   }
 
   public addAncestor(id: string, ...events: ResolverEvent[]): void {
     events.forEach(event => {
       const ancestorID = entityId(event);
-      if (!this.cache[ancestorID]) {
+      if (this.cache.get(ancestorID) === undefined) {
         const newParent = createNode(ancestorID);
-        this.cache[ancestorID] = newParent;
-        // Uncomment this line to populate the node's parent field, it will build a tree like
-        // node.parent.parent.parent going up the ancestry chain
-        // this.cache[id].parent = newParent;
-
-        // This method just adds all the ancestors to the root node's ancestors array
-        // The ordering of the ancestors should be the closest parent is towards at the front of the array
-        // the further the ancestor the more towards the end of the array
+        this.cache.set(ancestorID, newParent);
         if (!this.root.ancestors) {
           this.root.ancestors = [];
         }
         this.root.ancestors.push(newParent);
       }
-      this.cache[ancestorID].lifecycle.push(event);
+      const currentAncestor = this.cache.get(ancestorID);
```

Comment:
I think we could move this up into the previous `if` statement and remove the `if (currentAncestor !== undefined) {` check right?